regcomp.c

```diff
@@ -20712,18 +20541,10 @@ S_change_engine_size(pTHX_ RExC_state_t *pRExC_state, const Ptrdiff_t size)
     if (size > 0) {
         Zero(REGNODE_p(RExC_emit), size, regnode);
     }
-
-#ifdef RE_TRACK_PATTERN_OFFSETS
-    Renew(RExC_offsets, 2*RExC_size+1, U32);
-    if (size > 0) {
-        Zero(RExC_offsets + 2*(RExC_size - size) + 1, 2 * size, U32);
-    }
-    RExC_offsets[0] = RExC_size;
-#endif
 }
 
 STATIC regnode_offset
-S_regnode_guts(pTHX_ RExC_state_t *pRExC_state, const U8 op, const STRLEN extra_size, const char* const name)
+S_regnode_guts(pTHX_ RExC_state_t *pRExC_state, const STRLEN extra_size)
 {
     /* Allocate a regnode for 'op', with 'extra_size' extra (smallest) regnode
      * equivalents space.  It aligns and increments RExC_size
```

Comment:
You've removed 'op'; if you can make more sense of `with 'extra_size' extra (smallest) regnode equivalents space` that would help too. I'd suggest (based on my best guess): `Allocate a regnode that is (1 + extra_size) times as big as the smallest regnode`.